server/slack_notifier_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/applikatoni/applikatoni/models"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestNotifySlack(t *testing.T) {
+	target := &models.Target{Name: "staging"}
+
+	application := &models.Application{
+		GitHubOwner: "shipping-co",
+		GitHubRepo:  "main-web-app",
+	}
+
+	deployment := &models.Deployment{
+		ApplicationName: "web",
+		TargetName:      target.Name,
+		Branch:          "master",
+		CommitSha:       "f00b4r",
+	}
+
+	expectedMessage := slackMsg{
+		Username:  "Applikatoni Bot",
+		IconEmoji: ":it:",
+		Text: fmt.Sprintf("Successfully deployed %s :tada:\n to: %s \n on: %s %s",
+			deployment.ApplicationName,
+			deployment.TargetName,
+			deployment.Branch,
+			deployment.CommitSha),
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, _ := ioutil.ReadAll(r.Body)
+		expectedJson, _ := json.Marshal(expectedMessage)
+		if string(body) != string(expectedJson) {
+			t.Errorf("sent wrong payload expected=%v got=%v", expectedMessage, string(body))
```

Comment:
What do you mean exactly? I changed the `expectedMessage` in this test and got it to fail locally:

```
=== RUN TestNotifySlack
2015/12/03 07:06:25 Successfully notified Slack about deployment of web on staging, f00b4r!
--- FAIL: TestNotifySlack (0.00s)
        slack_notifier_test.go:43: sent wrong payload expected={Successfully foobar web :tada:
                 to: staging
                 on: master f00b4r Applikatoni Bot :it:} got={"text":"Successfully deployed web :tada:\n to: staging \n on: master f00b4r","username":"Applikatoni Bot","icon_emoji":":it:"}
FAIL
exit status 1
```
It does output the "successfully notified" message but that is because we don't return a non-200 response code when the expectation doesn't match the request body (adding this would be a nice touch, now that I think about it)